support-frontend/assets/pages/digital-subscription-landing/components/heroGift/heroStyles.js

```diff
@@ -0,0 +1,462 @@
+  -webkit-animation-fill-mode: both;
+  animation-delay: 1s;
+  -webkit-animation-delay: 1s;
+
+  animation-fill-mode: both;
+  -webkit-animation-fill-mode: both;
+  animation-delay: 1s;
+  -webkit-animation-delay: 1s;
+
+  @media (prefers-reduced-motion) {
+    animation: none;
+    -webkit-animation: none;
+    border-right: none;
+    -webkit-border-right: none;
+  }
+
+  @keyframes blink-caret-to {
+    from, to { border-color: transparent }
+    50% { border-color: white; }
+  }
+`;
+
+export const fromMeTyping = css`
+  overflow: hidden; /* Ensures the content is not revealed until the animation */
+  white-space: nowrap; /* Keeps the content on a single line */
+  letter-spacing: 0.01em; /* Adjust as needed */
+  margin-left: ${space[2]}px;
+  animation:
+    typing-from 0.5s steps(2, end);
+
+  /* This is to make it work on iPhones */
+  -webkit-animation-name: typing-from;
+  -webkit-animation-duration: 0.5s;
+  -webkit-animation-timing-function: steps(2, end);
+
+  animation-fill-mode: both;
+  -webkit-animation-fill-mode: both;
+  animation-delay: 2.4s;
+  -webkit-animation-delay: 2.4s;
+
+  color: ${brandAltBackground.primary};
+
+  @media (prefers-reduced-motion) {
+    animation: none;
+    border-right: none;
+  }
+
+  @keyframes typing-from {
+    from { width: 0 }
+    to { width: 14% }
+  }
+
+  @-webkit-keyframes typing-from {
+    from { width: 0 }
+    to { width: 14% }
+  }
+
+  ${headline.small({ fontWeight: 'bold' })};
+
+  ${from.mobileMedium} {
+    @keyframes typing-from {
+      from { width: 0 }
+      to { width: 12% }
+    }
+  }
+
+  ${from.mobileLandscape} {
+    ${headline.medium({ fontWeight: 'bold' })};
+    @keyframes typing-from {
+      from { width: 0 }
+      // Looks like a repetition but seems only to work at this size if specified
+      to { width: 12% }
+    }
+  }
+
+  ${from.tablet} {
+    @keyframes typing-from {
+      from { width: 0 }
+      to { width: 14% }
+    }
+  }
+
+  ${from.desktop} {
+    ${headline.large({ fontWeight: 'bold' })};
+    margin-left: 8px;
+    @keyframes typing-from {
+      from { width: 0 }
+      to { width: 13% }
+    }
+  }
+
+  ${from.leftCol} {
+    @keyframes typing-from {
+      from { width: 0 }
+      to { width: 15.5% }
+    }
+  }
+
+  ${from.wide} {
+    @keyframes typing-from {
+      from { width: 0 }
+      to { width: 13.25% }
+    }
+  }
+
+`;
+
+export const fromMeCursor = css`
+  overflow: hidden; /* Ensures the content is not revealed until the animation */
+  border-right: 1px solid white; /* The typwriter cursor */
+  animation:
+    blink-caret-from 0.7s steps(2, jump-end);
+
+  /* This is to make it work on iPhones */
+  -webkit-animation-name: blink-caret-from;
+  -webkit-animation-duration: 0.7s;
+  -webkit-animation-timing-function: steps(2, jump-end);
+
+  animation-fill-mode: both;
+  -webkit-animation-fill-mode: both;
+  animation-delay: 2.4s;
+  -webkit-animation-delay: 2.4s;
+
+  @media (prefers-reduced-motion) {
```

Comment:
I think it would be preferable to have no animation be the default, and to have the animation inside a `prefers-reduced-motion: no-preference` query. That way it shouldn't run on older devices that don't allow people to set a preference, which also don't tend to be that great at rendering CSS animations anyway.

Reply:
I think that you're right, but I'm wondering whether putting the animations inside an mq is harder to understand for someone trying to understand the code later? Definitely happy to discuss this further if you feel strongly that it should be this way.

Reply:
Yeah, it's maybe not the most straightforward to understand... how about pulling the actual query condition out into a variable with a clear name, which we can then reference in the actual CSS like we do with screen size queries, so it's a lot more obvious what the intent is?

```
const allowsAnimation = '@media (prefers-reduced-motion: no-preference)'

${allowsAnimation} {
	// animation CSS goes here
}
```